In [1]:
import pandas as pd

# bsd: bike-sharing-demand
bsd_train_path = '../../data/data_small/bike-sharing-demand/train.csv'
bsd_df = pd.read_csv(bsd_train_path)

bsd_df['datetime'] = pd.to_datetime(bsd_df['datetime'])

bsd_df['year'] = bsd_df['datetime'].dt.year
bsd_df['month'] = bsd_df['datetime'].dt.month
bsd_df['day'] = bsd_df['datetime'].dt.day
bsd_df['time_of_day'] = bsd_df['datetime'].dt.time
bsd_df.drop('datetime', axis=1, inplace=True)

bsd_df

season  holiday  workingday  weather   temp   atemp  humidity  \
0           1        0           0        1   9.84  14.395        81   
1           1        0           0        1   9.02  13.635        80   
2           1        0           0        1   9.02  13.635        80   
3           1        0           0        1   9.84  14.395        75   
4           1        0           0        1   9.84  14.395        75   
...       ...      ...         ...      ...    ...     ...       ...   
10881       4        0           1        1  15.58  19.695        50   
10882       4        0           1        1  14.76  17.425        57   
10883       4        0           1        1  13.94  15.910        61   
10884       4        0           1        1  13.94  17.425        61   
10885       4        0           1        1  13.12  16.665        66   

       windspeed  casual  registered  count  year  month  day time_of_day  
0         0.0000       3          13     16  2011      1    1    00:00:00  
1         0.0000       8          32     40  2011      1    1    01:00:00  
2         0.0000       5          27     32  2011      1    1    02:00:00  
3         0.0000       3          10     13  2011      1    1    03:00:00  
4         0.0000       0           1      1  2011      1    1    04:00:00  
...          ...     ...         ...    ...   ...    ...  ...         ...  
10881    26.0027       7         329    336  2012     12   19    19:00:00  
10882    15.0013      10         231    241  2012     12   19    20:00:00  
10883    15.0013       4         164    168  2012     12   19    21:00:00  
10884     6.0032      12         117    129  2012     12   19    22:00:00  
10885     8.9981       4          84     88  2012     12   19    23:00:00  

[10886 rows x 15 columns]

# Bike Sharing Demand Database
https://www.kaggle.com/competitions/bike-sharing-demand/data

## Descripción del dataset
El **Bike Sharing Demand** es un conjunto de datos creado para una competencia en Kaggle basado en datos reales del sistema de alquiler de bicicletas Capital Bikeshare en Washington D.C. entre 2011 y 2012. El dataset contiene está dividido en dos partes: el conjunto de entrenamiento (train.csv) con 10,886 filas y 15 variables, y el conjunto de prueba (test.csv) con 6,493 filas y 9 columnas. El objetivo principal es predecir la demanda de bicicletas por hora (count) en función de factores como el clima, la fecha y el tipo de día. 

Cada fila representa una hora específica, con atributos que describen las condiciones temporales (e.g., fecha, hora), climáticas (e.g., temperatura, clima) y la cantidad de alquileres (count, dividido en usuarios casuales y registrados). Este dataset es ideal para explorar patrones de movilidad urbana, predecir demanda y analizar influencias ambientales, ofreciendo oportunidades para aplicar técnicas de regresión, series temporales y visualización en un contexto comercial y analítico relevante.

#### Posibles análisis

1. Predicción de la Demanda de Bicicletas por Hora  
   
   Variables potencialmente interesantes: 
     - **count**: Variable objetivo numérica, suma de alquileres casuales y registrados, a predecir.  
     - **datetime (y derivados: hour, weekday)**: Hora y día de la semana, capturan patrones horarios y semanales de uso.  
     - **temp**: Temperatura en grados Celsius, afecta la disposición a usar bicicletas.  
     - **weather**: Condición climática (e.g., Clear, Rain), un predictor clave de la demanda.  
     - **workingday**: Indica si es día laboral (1) o no (0), influye en patrones de uso.  

2. Análisis de Patrones de Uso por Condiciones Climáticas  
   
   Variables potencialmente interesantes:
     - **count**: Número de alquileres, base para comparación según clima.  
     - **weather**: Tipo de clima, permite segmentar demanda por condiciones (e.g., lluvia vs. soleado).  
     - **temp**: Temperatura, mide efectos térmicos en el uso.  
     - **humidity**: Humedad relativa, puede influir en la comodidad de los usuarios.  

3. Segmentación de Horas por Patrones de Demanda  
   
   Variables potencialmente interesantes:
     - **count**: Demanda total, define el tamaño del uso por hora.  
     - **hour**: Hora del día, captura picos diarios (e.g., mañana vs. noche).  
     - **weekday**: Día de la semana, diferencia entre días laborales y fines de semana.  
     - **holiday**: Indica feriados, útil para segmentar días especiales.  

4. Análisis de Tendencias Temporales en la Demanda de Bicicletas
   
   Variables potencialmente interesantes:
     - **count**: Demanda de alquileres, base para analizar tendencias.  
     - **year**: Año (2011 o 2012), muestra cambios interanuales.  
     - **month**: Mes, captura estacionalidad (e.g., verano vs. invierno).  
     - **hour**: Hora, permite análisis intradía dentro de tendencias más amplias.